### PR TITLE
feat(encoder): add fastica to numeric encoders

### PR DESCRIPTION
**Pull request by JamesTang616**
feat(encoder): add fastica to numeric encoders, including related unit tests.

### CODE REVIEW

**Comment by nan-wang** [comment]
recheckcla

File: jina/executors/encoders/numeric/ica.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
+__copyright__ = "Copyright (c) 2020 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
```
Comment (nan-wang) [inline jina/executors/encoders/numeric/ica.py:3]: The bot will add this automatically.
```suggestion

```

File: jina/executors/encoders/numeric/ica.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    @batching
+    def train(self, data: 'np.ndarray', *args, **kwargs):
+        num_samples, num_features = data.shape
+        if not self.num_features:
+            self.num_features = num_features
+        if num_samples < 5 * num_features:
+            self.logger.warning(
+                'the batch size (={}) is suggested to be 5 * num_features(={}) to provide a balance between '
+                'approximation accuracy and memory consumption.'.format(num_samples, num_features))
+        self.model._fit(data)
```
Comment (nan-wang) [inline jina/executors/encoders/numeric/ica.py:54]:
```suggestion
        self.model.fit(data)

```

**Comment by hanxiao** [comment]
bad PR CLA can't be triggered 

https://github.com/jina-ai/jina/pull/508#issuecomment-640166286

image

closed and resubmit PR please

**Pull request closed without merging**